pyparsing/core.py

```diff
@@ -1261,33 +1264,34 @@ def scan_string(
                 except ParseException:
                     loc = preloc + 1
                 else:
-                    if nextLoc > loc:
```

Comment:
Please move this to a separate PR, so it can be tested in isolation from all the other more-cosmetic changes.